src/indexer.c

```diff
@@ -231,6 +231,9 @@ static void indexBulkFields(RSAddDocumentCtx *aCtx, RedisSearchCtx *sctx) {
       }
 
       if (IndexerBulkAdd(bulk, cur, sctx, doc->fields + ii, fs, fdata, &cur->status) != 0) {
+        IndexError_AddError(&cur->spec->stats.indexError, cur->status.detail, doc->docKey);
+        IndexError_AddError(&cur->spec->fields[fs->index].indexError, cur->status.detail, doc->docKey);
```

Comment:
Doc and AddCtx are aligned, but not necessarily with the spec if some fields are missing. Instead of using the loop index that corresponds to the field index in the doc and AddCtx, we want the index of the field. Luckily the field stores its own index so we just need use it instead of the loop index